ops/model.py

```diff
@@ -213,20 +199,48 @@ def __iter__(self):
         return iter(self._data)
 
     def __getitem__(self, relation_name):
+        is_peer = relation_name in self._peers
         relation_list = self._data[relation_name]
         if relation_list is None:
             relation_list = self._data[relation_name] = []
-            for relation_id in self._backend.relation_ids(relation_name):
-                relation_list.append(Relation(relation_name, relation_id, self._our_unit, self._backend, self._cache))
+            for rid in self._backend.relation_ids(relation_name):
+                relation = Relation(relation_name, rid, is_peer, self._our_unit, self._backend, self._cache)
+                relation_list.append(relation)
         return relation_list
 
+    def _get_unique(self, relation_name, relation_id=None):
+        if not isinstance(relation_id, (int, type(None))):
+            raise ModelError(f'relation name {relation_id} must be int or None not {type(relation_id).__name__}')
+        if relation_id is not None:
+            for relation in self[relation_name]:
+                if relation.id == relation_id:
+                    return relation
+            else:
+                # The relation may be dead, but it is not forgotten.
+                is_peer = relation_name in self._peers
+                return Relation(relation_name, relation_id, is_peer, self._our_unit, self._backend, self._cache)
+        else:
```

Comment:
this might be my recent time with Go showing: is the `else` necessary, here? I don't remember if that's pythonic or not :-D

Reply:
I suppose I could just drop it since every branch under the first if contains its own return statement or raises an exception.

Reply:
https://github.com/canonical/operator/pull/107/commits/6a058cb1cd04cb7f3d4ea38c01fd13a04d95cd03